### ISSUE TITLE
[BUG] expect(locator).toBeDisabled() returning false when it should return true

### ISSUE THREAD

**Issue by shtlrs** [microsoft/playwright#10734]
**Context:**
- Playwright Version: [1.17.1]
- Operating System: [Windows]
- Node.js version: [16.10.0]
- Browser: [Chromium]

**Code Snippet**

```typescript
import { expect, test } from "@playwright/test";

const cookie1 = { name: "_iub_cs-152080", path: "/", domain: ".expensya.com", httpOnly: false, value: "%7B%22timestamp%22%3A%222021-09-30T13%3A39%3A08.712Z%22%2C%22version%22%3A%221.33.0%22%2C%22purposes%22%3A%7B%221%22%3Atrue%2C%222%22%3Atrue%2C%223%22%3Atrue%2C%224%22%3Atrue%2C%225%22%3Atrue%7D%2C%22id%22%3A152080%7D" };
const cookie2 = { name: "_iub_cs-422042", path: "/", domain: ".expensya.com", httpOnly: false, value: "%7B%22timestamp%22%3A%222021-11-23T10%3A34%3A24.714Z%22%2C%22version%22%3A%221.34.0%22%2C%22purposes%22%3A%7B%221%22%3Atrue%2C%222%22%3Atrue%2C%223%22%3Atrue%2C%224%22%3Atrue%2C%225%22%3Atrue%7D%2C%22id%22%3A422042%2C%22cons%22%3A%7B%22rand%22%3A%22915b88%22%7D%7D" };
const cookie3 = { name: "_iub_cs-56424840", path: "/", domain: ".expensya.com", httpOnly: false, value: "%7B%22timestamp%22%3A%222021-11-23T10%3A51%3A02.098Z%22%2C%22version%22%3A%221.34.0%22%2C%22purposes%22%3A%7B%221%22%3Atrue%2C%222%22%3Atrue%2C%223%22%3Atrue%2C%224%22%3Atrue%2C%225%22%3Atrue%7D%2C%22id%22%3A56424840%2C%22cons%22%3A%7B%22rand%22%3A%2263846b%22%7D%7D" };
const cookie4 = { name: "_iub_cs-56499786", path: "/", domain: ".expensya.com", httpOnly: false, value: "%7B%22timestamp%22%3A%222021-10-20T13%3A42%3A54.860Z%22%2C%22version%22%3A%221.33.1%22%2C%22purposes%22%3A%7B%221%22%3Atrue%2C%222%22%3Atrue%2C%223%22%3Atrue%2C%224%22%3Atrue%2C%225%22%3Atrue%7D%2C%22id%22%3A56499786%2C%22cons%22%3A%7B%22rand%22%3A%221d509b%22%7D%7D" };

/**
 * Injects the Iuebenda cookies to prevent the modal from popping up when the user's locale is
 * different than the account's language
 */
export async function injectIuebendaCookies(page) {
    let cookies = await page.context().cookies();
    cookies = cookies.filter(cookie => cookie.name.includes("iub"));
    if (cookies.length === 0) {
        await page.click('button[class="iubenda-cs-accept-btn iubenda-cs-btn-primary"]');
        await page.context().addCookies([cookie1, cookie2, cookie3, cookie4]);
    }
}


test('Test description', async ({ page }) => {

    await page.goto("https://dev.expensya.com");
    await injectIuebendaCookies(page);
    await page.fill("#email","adRO@yopmail.com");
    await page.fill('[name="Password"]', "a");
    await page.click("#login-btn");
    await page.click("#admin_navbar");
    await page.click('[href="#/Admin/Contact"]');
    await page.click(":has-text('ADMIN')");
    expect(page.locator("#firstName")).toBeDisabled();

});
```

**Describe the bug**

Playwright indicates that an input element is not disabled whereas it is.
I've tried the browser's console and when I query the element and then log its disabled attribute to console, it shows true.
I also tried to do it step by step with playwright using python (REPL), and when I do page.is_disabled("#firstName"), it returns true.
But when using it with typescript, it just doesn't work
[trace.zip](https://github.com/microsoft/playwright/files/7662536/trace.zip)
.

**Comment by dgozman** [comment]
Did you forget to await? This test works for me with the following changes. Note that `":has-text('ADMIN')"` resolves to body, so I used a better selector.

```js
    // ...
    await page.click("text=ADMIN Admin");
    await expect(page.locator("#firstName")).toBeDisabled();
```

**Comment by shtlrs** [comment]
> Did you forget to await? This test works for me with the following changes. Note that `":has-text('ADMIN')"` resolves to body, so I used a better selector.
> 
> ```js
>     // ...
>     await page.click("text=ADMIN Admin");
>     await expect(page.locator("#firstName")).toBeDisabled();
> ```

My bad, I only missed the await in this sample.
But I have it in my code :s

And I'm having issues uploading the file to github (I'm getting "is not included in the list), do you want me to send it over on slack ?

**Comment by dgozman** [comment]
> And I'm having issues uploading the file to github (I'm getting "is not included in the list), do you want me to send it over on slack ?

Sure.

**Comment by dgozman** [comment]
Thank you for the trace file. The problem here is that `<input>` is inside a disabled `<fieldset>`, and Playwright did not pick it up. This was addressed by #9927, so will be fixed in v1.18.
